### PR TITLE
Implement integration test script runner

### LINKED ISSUE #901: Add CI test to verify JPN IXDS manifest compatibility
### What should we change and why?

Recent IXDS changes have broken the JPN IXDS implementation twice now (see #841). We should add a test to CI to verify the manifest file can be loaded and an instance file extracted.

### PR DESCRIPTION
**Pull request by aaroncameron-wk**
#### Reason for change
There will be a need for running general integration test scripts, separate from the conformance suite pattern we currently use. These integration tests should be as flexible as possible.

#### Description of change
- Implemented standalone `download_cache.py` script for downloading public caches in a cross platform way (to remove the need separate conditional steps in workflows)
- Created script running framework in `tests/integration_tests/scripts` based off of the existing conformance suite running framework
- Created `requirements-plugins.txt` requirements set for integration tests and builds
- Implemented test for Japan IXDS (Resolves #901)
- Implemented test for triggering `ixbrl-viewer` from the command line (to help test #734)
- Created workflow to run integration test scripts on pull requests
  - Added dispatch-only version that runs the tests in parallel 
- Added steps to build workflows to run these tests against builds

#### Steps to Test
Should pass on this PR (pull_request trigger):
- [ ] test-scripts.yml

Run on fork (see recent runs on my fork [here](https://github.com/aaroncameron-wk/arelle-public/actions)):
- [ ] test-scripts-parallel.yml
- [ ] conformance-suites.yml
- [ ] build-macos.yml (Note: codesign/notarize can now be optionally skipped)
- [ ] build-windows.yml
- [ ] build-linux.yml

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: requirements-integration.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ixbrl-viewer==1.4.10
```
Comment (austinmatherne-wk) [inline requirements-integration.txt:1]: Do you have plans to align this with #990 so we don't have this dependency in two different files? 
Maybe name this `requirements-plugins.txt` and drop `-r requirements-dev.txt` so it can be required by `requirements-build.txt`?

File: .github/workflows/build-macos.yml [hunk 2]
```diff
@@ -201,6 +213,17 @@ jobs:
           ditto -c -k --keepParent "build/Arelle.app" "notarization.zip"
           xcrun notarytool submit "notarization.zip" -v --apple-id "$USERNAME" --password "$PASSWORD" --team-id "$TEAM_ID" --wait --timeout 30m --output-format json
           xcrun stapler staple -v "build/Arelle.app"
+      - name: "[Test] Test build artifact"
+        run : |
+          pip install -r requirements-integration.txt
```
Comment (austinmatherne-wk) [inline .github/workflows/build-macos.yml:218]: Why is the pip install required, isn't the plugin loaded from the build?

File: arelle/PluginManager.py [hunk 3]
```diff
@@ -331,9 +331,9 @@ def parsePluginInfo(moduleURL: str, moduleFilename: str, entryPoint: EntryPoint
                 if entryPoint:
                     moduleInfo["moduleURL"] = moduleFilename  # pip-installed plugins need absolute filepath
                     moduleInfo["entryPoint"] = {
-                        "module": entryPoint.module,
+                        "module": getattr(entryPoint, 'module', None),
```
Comment (austinmatherne-wk) [inline arelle/PluginManager.py:334]: Why why this change needed?

File: tests/integration_tests/scripts/tests/ixbrl-viewer_cli.bat [hunk 4]
```diff
@@ -0,0 +1,31 @@
+@REM Download samples:
```
Comment (austinmatherne-wk) [inline tests/integration_tests/scripts/tests/ixbrl-viewer_cli.bat:1]: Out of curiosity, did you look into using a python script to handle what's in the batch and shell script files and only using a subprocess to call the arelle command line?

File: .github/workflows/build-linux.yml [hunk 5]
```diff
@@ -145,6 +145,25 @@ jobs:
...
+        with:
+          cache: 'pip'
+          check-latest: true
+          python-version: ${{ inputs.python_version }}
+      - name: "[Test] Test build artifact"
+        run : |
+          mkdir .test
+          tar -xvzf "${{ env.BUILD_ARTIFACT_PATH }}" -C .test
+          pip install -r requirements-integration.txt
+          python3 tests/integration_tests/download_cache.py --name "all_scripts.zip"
```
Comment (austinmatherne-wk) [inline .github/workflows/build-linux.yml:159]: nit: we don't use `python3` elsewhere

File: .github/workflows/conformance-suites.yml [hunk 6]
```diff
@@ -100,28 +100,6 @@ jobs:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_KEY }}
           aws-region: us-east-1
-      - name: Download XBRL validation config
```
Comment (austinmatherne-wk) [inline .github/workflows/conformance-suites.yml:103]: Nice!

**Review by austinmatherne-wk — APPROVED** [review]
